python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -340,8 +329,38 @@ def _messages_thread(
                     break
             time.sleep(1)
 
+    def _logs_thread(
+        self,
+        params: PipesParams,
+        is_task_complete: Event,
+    ) -> None:
+        # Start once we have received the opened message from the external process
+        while self.opened_payload is None:
+            time.sleep(5)
+
+        log_params = {**params, **self.opened_payload}
+        try:
+            for log_reader in self.log_readers:
+                log_reader.start(log_params, is_task_complete)
+            while True:
+                if is_task_complete.is_set():
+                    break
+                else:
+                    time.sleep(1)
```

Comment:
We can't really use a timeout though right, since we don't know how long the job takes?

I just put `is_session_closing.wait()`.

Reply:
you can do `while not event.wait(N):`  if you have something in the loop to do every N seconds but since you don't here agree just `wait()` is right